### PR TITLE
Deprecation info api for camel case date formats

### PR DESCRIPTION
**Pull request by pgomulka**
using camel case date formats was deprecaetd in
https://github.com/elastic/elasticsearch/pull/59949
but only using deprecation logger.
This commit adds deprecation info checks

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by pgomulka** [comment]
awaits https://github.com/elastic/elasticsearch/pull/83841 for the link to breaking changes doc

File: docs/changelog/83839.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pr: 83839
+summary: Deprecation info api for camel case date formats
+area: Infra/Core
+type: deprecation
+issues: []
```
Comment (pgomulka) [inline docs/changelog/83839.yaml:5]: it previously had this generated too.
Not sure this was correct - hard to remove it to get CI to pass
```
deprecation:
title: Deprecation info api for camel case date formats
area: Infra/Core
details: Please describe the details of this change for the release notes. You can
use asciidoc.
impact: Please describe the impact of this change to users

```

Comment (pgomulka) [reply docs/changelog/83839.yaml:5]: https://github.com/elastic/elasticsearch/pull/83839#issuecomment-1036372110
but it fails with..
```
[validate YAML][ERROR][83839.yaml][$.deprecation.area: does not have a value in the enumeration [API, Aggregation, Allocation, Authentication, CCR, Cluster, Discovery, Engine, HTTP, Highlighters, Indices, Java, License Information, Logging, Machine Learning, Mappings, Networking, Packaging, Plugins, Script Cache, Search Changes, Search, Security, Settings, Snapshot and Restore, Transform, Transport]]
[validate YAML][ERROR][83839.yaml][$.deprecation.body: is missing but it is required]
[validate YAML][ERROR][83839.yaml][$.deprecation.details: is not defined in the schema and the schema does not allow additional properties]
[validate YAML][ERROR][83839.yaml][$.deprecation.impact: is not defined in the schema and the schema does not allow additional properties]


```

Comment (grcevski) [reply docs/changelog/83839.yaml:5]: I think the details multi-line text might have issues with YML. I think you'll need ` > ` after `details:` and two space indent for each line.

**Review by grcevski — APPROVED** [review]
Change looks good to me. Thanks!

**Pull request merged**